src/tests/ftest/io/llnl_mpi4py.py

```diff
@@ -0,0 +1,157 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+    The Government's rights to use, modify, reproduce, release, perform,
+    display, or disclose this software are subject to the terms of the Apache
+    License as provided in Contract No. B609815.
+    Any reproduction of computer software, computer software documentation, or
+    portions thereof marked with this legend must also reproduce the markings.
+    '''
+
+from __future__    import print_function
+import os
+import sys
+import json
+from avocado       import Test
+
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+import ServerUtils
+import WriteHostFile
+from MpioUtils import MpioUtils, MpioFailed
+from daos_api import DaosContext, DaosPool, DaosApiError
+
+class LlnlMpi4py(Test):
+    """
+    Runs LLNL and MPI4PY test suites.
+    """
+
+    def __init__(self, *args, **kwargs):
+
+        super(LlnlMpi4py, self).__init__(*args, **kwargs)
+
+        self.basepath = None
+        self.server_group = None
+        self.context = None
+        self.pool = None
+        self.mpio = None
+        self.hostlist_servers = None
+        self.hostfile_servers = None
+        self.hostlist_clients = None
+        self.hostfile_clients = None
+
+    def setUp(self):
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as var_file:
+            build_paths = json.load(var_file)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+
+        self.server_group = self.params.get("name", '/server/', 'daos_server')
+
+        # setup the DAOS python API
+        self.context = DaosContext(build_paths['PREFIX'] + '/lib/')
+
+        self.hostlist_servers = self.params.get("test_servers", '/run/hosts/')
+        self.hostfile_servers = WriteHostFile.WriteHostFile(self.hostlist_servers,
+                                                            self.workdir)
+        print("Host file servers is: {}".format(self.hostfile_servers))
+
+        self.hostlist_clients = self.params.get("test_clients", '/run/hosts/')
+        self.hostfile_clients = WriteHostFile.WriteHostFile(self.hostlist_clients,
+                                                            self.workdir, None)
+        print("Host file clients is: {}".format(self.hostfile_clients))
+
+        # start servers
+        ServerUtils.runServer(self.hostfile_servers, self.server_group,
```

Comment:
Agent bin need to start before starting server

Reply:
Will do.